openid-sse-framework-1_0.txt

```diff
@@ -772,434 +749,358 @@ Tulshibagwale, et al.   Expires December 10, 2021              [Page 13]
+      ],
+    "delivery": {
+      "delivery_method":
+        "https://schemas.openid.net/secevent/risc/delivery-method/push",
+      "url": "https://receiver.example.com/events"
+    },
+    "events_supported": [
+      "urn:example:secevent:events:type_1",
+      "urn:example:secevent:events:type_2",
+      "urn:example:secevent:events:type_3"
+    ],
+    "events_requested": [
+      "urn:example:secevent:events:type_2",
+      "urn:example:secevent:events:type_3",
+      "urn:example:secevent:events:type_4"
+    ],
+    "events_delivered": [
+      "urn:example:secevent:events:type_2",
+      "urn:example:secevent:events:type_3"
+    ]
+  }
 
-   POST /sse/status HTTP/1.1
-   Host: transmitter.example.com
-   Authorization: Bearer eyJ0b2tlbiI6ImV4YW1wbGUifQo=
+                Figure 15: Example: Create Stream Response
 
-   {
-     "status": "paused"
-   }
+   Errors are signaled with HTTP status codes as follows:
 
+    +======+=========================================================+
+    | Code | Description                                             |
+    +======+=========================================================+
+    | 400  | if the request cannot be parsed                         |
+    +------+---------------------------------------------------------+
+    | 401  | if authorization failed or it is missing                |
+    +------+---------------------------------------------------------+
+    | 403  | if the Event Receiver is not allowed to create a stream |
+    +------+---------------------------------------------------------+
 
-     Figure 18: Example: Update Stream Status Request Without Optional
-                                  Fields
+                      Table 1: Create Stream Errors
 
-   The following is a non-normative example of an Update Stream Status
-   request with optional fields:
 
-   POST /sse/status HTTP/1.1
-   Host: transmitter.example.com
-   Authorization: Bearer eyJ0b2tlbiI6ImV4YW1wbGUifQo=
 
-   {
-     "status": "paused",
-     "subject": {
-       "tenant" : {
-         "format" : "iss_sub",
-         "iss" : "http://example.com/idp1",
-         "sub" : "1234"
-       }
-     },
-     "reason": "Disabled by administrator action."
-   }
 
 
-   Figure 19: Example: Update Stream Status Request With Optional Fields
+Tulshibagwale, et al.        Standards Track                   [Page 18]
+
+                          openid-sse-framework                 June 2021
 
-   The following is a non-normative example response:
 
+7.1.1.2.  Reading a Stream's Configuration
 
+   An Event Receiver gets the current configuration of a stream by
+   making an HTTP GET request to the Configuration Endpoint.  On
+   receiving a valid request the Event Transmitter responds with a 200
+   OK response containing a JSON [RFC7159] representation of the
+   stream's configuration in the body.
 
+   The GET request MUST include the "stream_id" as a parameter in order
```

Comment:
We already have a section of the spec devoted to backwards compatibility for Google, and they could use that to maintain the existing behavior. If we did make `stream_id` optional, then I would expect this function to return a list of all streams. That's not a bad idea, but it's not backwards compatible.

Reply:
I have made `stream_id` optional and described what should happen in the case of no streams, one stream, and many streams. I think it is probably wise to provide Receivers with a way to get all of the stream in case they lose track of stream ids on their end.